apis/v1alpha2/grpcroute_types.go

```diff
@@ -549,6 +549,42 @@ type GRPCRouteFilter struct {
 }
 
 // GRPCBackendRef defines how a GRPCRoute forwards a gRPC request.
+//
+// Note that when a namespace different than the local namespace is specified, a
+// ReferenceGrant object is required in the referent namespace to allow that
+// namespace's owner to accept the reference. See the ReferenceGrant
+// documentation for details.
+//
+// <gateway:experimental:description>
+//
+// When the BackendRef points to a Kubernetes Service, implementations SHOULD honor the
+// appProtocol field if it is set for the target Service Port.
+//
+// Implementations supporting appProtocol MUST recognize the Kubernetes Standard Application Protocols
+// defined in [KEP-3726]. This supports IANA standard service names and extra constants defined
+// in the KEP that have a prefix of "kubernetes.io/". Gateway API MAY define additional
+// constants with the prefix "gateway.networking.k8s.io/"
+//
+// If a Service appProtocol isn't specified an implementation MAY infer the backend
+// protocol through its own means. Implementations MAY infer the protocol from the
+// Route type referring to the backend Service.
+//
+// Implementations MAY support Kubernetes Service BackendRefs that are multiplexing TCP
+// and UDP on the same port. Otherwise implementations MUST set the Route ResolvedRefs
+// condition to False with the "UnsupportedProtocol" Reason with a clear message that
+// multiplexing is not supported.
```

Comment:
I've generally thought that the Route implies the protocol that should be used to connect to a backend. I'd rather leave this for a follow up issue/discussion as I think there's a lot of nuance here that we might be missing.

Reply:
removed